src/Common/src/System/Globalization/IdnMapping.Unix.cs

```diff
@@ -67,5 +73,30 @@ private uint Flags
                 return (uint)flags;
             }
         }
+
+        /// <summary>
+        /// ICU doesn't check for invalid characters unless the STD3 rules option
+        /// is enabled.
+        ///
+        /// To match Windows behavior, we walk the string ourselves looking for these
+        /// bad characters so we can continue to throw ArgumentException in these cases. 
+        /// </summary>
+        private static void CheckInvalidIdnCharacters(string s, uint flags, string paramName)
+        {
+            if ((flags & Interop.GlobalizationNative.UseStd3AsciiRules) == 0)
+            {
+                for (int i = 0; i < s.Length; i++)
```

Comment:
I'd look at changing the check from `(c <= 0x1F || c == 0x7F)` to `(c <= 0x1F | c == 0x7F)` and removing the branch that only short-circuits on an equality-check (and that will be hit in the more common case anyway) before looking at unrolling.